src/main/java/com/sonyericsson/hudson/plugins/gerrit/trigger/workflow/GerritReviewStep.java

```diff
@@ -0,0 +1,144 @@
+ * THE SOFTWARE.
+ */
+package com.sonyericsson.hudson.plugins.gerrit.trigger.workflow;
+
+import com.sonyericsson.hudson.plugins.gerrit.trigger.gerritnotifier.ToGerritRunListener;
+import hudson.Extension;
+import hudson.Util;
+import hudson.model.Run;
+import javax.annotation.CheckForNull;
+import javax.inject.Inject;
+import org.jenkinsci.plugins.workflow.steps.AbstractStepDescriptorImpl;
+import org.jenkinsci.plugins.workflow.steps.AbstractStepImpl;
+import org.jenkinsci.plugins.workflow.steps.AbstractSynchronousStepExecution;
+import org.jenkinsci.plugins.workflow.steps.StepContextParameter;
+import org.kohsuke.stapler.DataBoundConstructor;
+import org.kohsuke.stapler.DataBoundSetter;
+
+/**
+ * Allows altering the Gerrit review posted at the end of build during the build.
+ * @author Teemu Murtola &lt;teemu.murtola@gmail.com&gt
+ */
+public class GerritReviewStep extends AbstractStepImpl {
+
+    private String customUrl;
+    private String unsuccessfulMessage;
+
+    /**
+     * Constructor.
+     *
+     * There are no mandatory parameters to the step.
+     */
+    @DataBoundConstructor
+    public GerritReviewStep() {
+    }
+
+    /**
+     * Gets the custom URL for a step.
+     * @return the URL.
+     */
+    @CheckForNull
+    public String getCustomUrl() {
+        return customUrl;
+    }
+
+    /**
+     * Sets a custom URL to post for a build.
+     * @param customUrl the URL to post.
+     */
+    @DataBoundSetter
+    public void setCustomUrl(String customUrl) {
+        this.customUrl = Util.fixEmptyAndTrim(customUrl);
+    }
+
+    /**
+     * Gets the unsuccessful message for a step.
+     * @return the message.
+     */
+    @CheckForNull
+    public String getUnsuccessfulMessage() {
+        return unsuccessfulMessage;
+    }
+
+    /**
+     * Sets additional information to post for a failed build.
+     * @param unsuccessfulMessage Additional information to post for a failed build.
+     */
+    @DataBoundSetter
+    public void setUnsuccessfulMessage(String unsuccessfulMessage) {
+        this.unsuccessfulMessage = Util.fixEmptyAndTrim(unsuccessfulMessage);
+    }
+
+    /**
+     * Executes the GerritReviewStep.
+     */
+    public static class Execution extends AbstractSynchronousStepExecution<Void> {
+
+        @StepContextParameter
+        private transient Run build;
+
+        @Inject
+        private transient GerritReviewStep step;
+
+        @Override
+        protected Void run() throws Exception {
+            ToGerritRunListener listener = ToGerritRunListener.getInstance();
+            String customUrl = step.getCustomUrl();
+            if (customUrl != null) {
+                listener.setBuildCustomUrl(build, customUrl);
+            }
+            String unsuccessfulMessage = step.getUnsuccessfulMessage();
+            if (unsuccessfulMessage != null) {
+                listener.setBuildFailureMessage(build, unsuccessfulMessage);
```

Comment:
Ok, I can rename the method added here (if you think it would improve things), but leave the others intact. I assume that obtainFailureMessage() could be renamed without worries about compatibility (since it is private), but that still leaves the one existing method with different naming. I can't judge whether any other plugin could or should really depend on it.